### PR TITLE
Fix automatically declared parameters descriptor type

### PR DESCRIPTION
**Pull request by ivanpauno**
Fixes https://github.com/ros2/ros2/issues/1173.
Fixes https://github.com/ros2/rclpy/issues/829.

### CODE REVIEW

File: rclpy/rclpy/node.py [hunk 1]
```diff
@@ -205,17 +205,22 @@ def __init__(
...
-            self._descriptors.update({p: ParameterDescriptor() for p in self._parameters})
-
         # Clock that has support for ROS time.
         # Note: parameter overrides and parameter event publisher need to be ready at this point
         # to be able to declare 'use_sim_time' if it was not declared yet.
         self._clock = ROSClock()
         self._time_source = TimeSource(node=self)
         self._time_source.attach_clock(self._clock)
 
+        if automatically_declare_parameters_from_overrides:
```
Comment (adityapande-1995) [inline rclpy/rclpy/node.py:215]: According to the comment above : "parameter overrides and parameter event publisher need to be ready at this point". Should we move this section of code above the ``self._clock`` line, like it originally was ?

Comment (ivanpauno) [reply rclpy/rclpy/node.py:215]: parameters overrides and event publisher are ready when `ROSClock()` is called, even with these lines moved.
`Node.declare_parameters()` needs `self._clock`, so I had to switch the order.

**Comment by ivanpauno** [comment]
> Tested with rolling, resolves the issue. Need to backport this to galactic (which the issue references) as well. Maybe the mergify bot can help ?

Yes, we should use the mergify bot after merging this PR to master.

File: rclpy/rclpy/node.py (hunk 1 above)
Comment (adityapande-1995) [reply rclpy/rclpy/node.py:215]: Ah, makes sense

**Review by adityapande-1995 — APPROVED** [review]

**Comment by ivanpauno** [comment]
CI:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/15659/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/10337/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/13325/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/15882/)

**Comment by ivanpauno** [comment]
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/15898/)

**Comment by ivanpauno** [comment]
Same failures can be seen in windows repeated nightly jobs (e.g. [this one](https://ci.ros2.org/view/nightly/job/nightly_win_rep/2442/#showFailuresLink)).
Going in!

**Comment by ivanpauno** [comment]
@Mergifyio backport galactic

**Pull request merged**